tracing-subscriber/src/filter/targets.rs

```diff
@@ -0,0 +1,410 @@
+/// use tracing_subscriber::filter;
+///
+/// let filter = "my_crate=info,my_crate::interesting_module=trace,other_crate=debug"
+///     .parse::<filter::Targets>()?;
+///
+/// // The parsed filter is identical to a filter constructed using `with_target`:
+/// assert_eq!(
+///     filter,
+///     filter::Targets::new()
+///         .with_target("my_crate", Level::INFO)
+///         .with_target("my_crate::interesting_module", Level::TRACE)
+///         .with_target("other_crate", Level::DEBUG)
+/// );
+/// # Ok(()) }
+/// ```
+///
+/// This is particularly useful when the list of enabled targets is configurable
+/// by the user at runtime.
+///
+/// [target]: tracing_core::Metadata::target
+/// [level]: tracing_core::Level
+/// [`Filter`]: crate::layer::Filter
+/// [`Layer`]: crate::layer::Layer
+/// [plf]: crate::layer#per-layer-filtering
+/// [global]: crate::layer#global-filtering
+/// [filtering]: crate::layer#filtering-with-layers
+/// [`env_logger` crate]: https://docs.rs/env_logger/0.9.0/env_logger/index.html#enabling-logging
+/// [`EnvFilter`]: crate::filter::EnvFilter
+#[derive(Debug, Default, Clone)]
+pub struct Targets(DirectiveSet<StaticDirective>);
+
+impl Targets {
+    /// Returns a new `Targets` filter.
+    ///
+    /// This filter will enable no targets. Call [`with_target`] or [`with_targets`]
+    /// to add enabled targets, and [`with_default`] to change the default level
+    /// enabled for spans and events that didn't match any of the provided targets.
+    ///
+    /// [`with_target`]: Targets::with_target
+    /// [`with_targets`]: Targets::with_targets
+    /// [`with_default`]: Targets::with_default
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Enables spans and events with [target]s starting with the provided target
+    /// prefix if they are at or below the provided [`LevelFilter`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tracing_subscriber::filter;
+    /// use tracing_core::Level;
+    ///
+    /// let filter = filter::Targets::new()
+    ///     // Enable the `INFO` level for anything in `my_crate`
+    ///     .with_target("my_crate", Level::INFO)
+    ///     // Enable the `DEBUG` level for a specific module.
+    ///     .with_target("my_crate::interesting_module", Level::DEBUG);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [`LevelFilter::OFF`] can be used to disable a particular target:
+    /// ```
+    /// use tracing_subscriber::filter::{Targets, LevelFilter};
+    /// use tracing_core::Level;
+    ///
+    /// let filter = Targets::new()
+    ///     .with_target("my_crate", Level::INFO)
+    ///     // Disable all traces from `annoying_module`.
+    ///     .with_target("my_crate::interesting_module", LevelFilter::OFF);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [target]: tracing_core::Metadata::target
+    pub fn with_target(mut self, target: impl ToString, level: impl Into<LevelFilter>) -> Self {
```

Comment:
🤔 `ToString` as a bound feels very odd to me, I’d expect either `Into<String>` or `Dispaly`.